modules/rgbd/include/opencv2/rgbd/volume.hpp

```diff
@@ -50,9 +46,65 @@ enum class VolumeType
     HASHTSDF = 1
 };
 
-cv::Ptr<Volume> makeVolume(VolumeType _volumeType, float _voxelSize, cv::Affine3f _pose,
-                           float _raycastStepFactor, float _truncDist, int _maxWeight,
-                           float _truncateThreshold, Point3i _resolution);
+struct VolumeParams
+{
+    /** @brief Type of Volume
+        Values can be TSDF (single volume) or HASHTSDF (hashtable of volume units)
+    */
+    VolumeType type;
+
+    /** @brief Resolution of voxel space
+        Number of voxels in each dimension.
+    */
+    Vec3i resolution;
+
+    /** @brief Resolution of volumeUnit in voxel space
+        Number of voxels in each dimension for volumeUnit
+        Applicable only for hashTSDF.
+    */
+    int unitResolution = 0;
+
+    /** @brief Initial pose of the volume in meters */
+    cv::Affine3f pose;
```

Comment:
As far as I know Affine3f coerces the matrix into SE3, i.e., the R matrix has determinant always +1, so changing to Matx44f is not advisable I think.

Reply:
In fact nothing stops a user from spoiling `Affine3f` since:
* it can be constructed from arbitrary 4x4 matrix (or 3x3 R + 1x3 t matrices)
* it keeps the same `Matx44f` inside as a public member
So in both cases a user should be careful about the value.

Reply:
Done. Changed most interfaces into Matx44f. Personally though I think we should consider adding a proper SE3 pose class

Reply:
There are plans to add quaternions and possibly dual quaternions into OpenCV 5.0. If they are done properly, there will be consistent data structures to keep both SO(3) and SE(3) transforms.